### ISSUE TITLE
Scrub error

### ISSUE THREAD

**Issue by wgriffa757** [rockstor/rockstor-core#1762]
HI I received an email from my Rockstor saying....

Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib64/python2.7/logging/handlers.py", line 154, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib64/python2.7/logging/__init__.py", line 324, in getMessage
    msg = str(self.msg)
TypeError: __str__ returned non-string (type list)
Logged from file pool_scrub.py, line 78
Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib64/python2.7/logging/handlers.py", line 154, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib64/python2.7/logging/__init__.py", line 324, in getMessage
    msg = str(self.msg)
TypeError: __str__ returned non-string (type list)
Logged from file pool_scrub.py, line 39

Below is the contents of my scrub status file found in  /var/lib/btrfs

scrub status:1
20d37f57-4cf4-4528-afb8-c62527fb15fe:1|data_extents_scrubbed:0|tree_extents_scrubbed:0|data_bytes_scrubbed:0|tree_bytes_scrubbed:0|read_errors:0|csum_errors:0|verify_errors:0|no_csum:0|csum_discards:0|super_errors:0|malloc_errors:0|uncorrectable_errors:0|corrected_errors:0|last_physical:0|t_start:1499061605|t_resumed:0|duration:0|canceled:0|finished:1
20d37f57-4cf4-4528-afb8-c62527fb15fe:3|data_extents_scrubbed:0|tree_extents_scrubbed:0|data_bytes_scrubbed:0|tree_bytes_scrubbed:0|read_errors:0|csum_errors:0|verify_errors:0|no_csum:0|csum_discards:0|super_errors:0|malloc_errors:0|uncorrectable_errors:0|corrected_errors:0|last_physical:0|t_start:1499061605|t_resumed:0|duration:0|canceled:0|finished:1
20d37f57-4cf4-4528-afb8-c62527fb15fe:4|data_extents_scrubbed:0|tree_extents_scrubbed:0|data_bytes_scrubbed:0|tree_bytes_scrubbed:0|read_errors:0|csum_errors:0|verify_errors:0|no_csum:0|csum_discards:0|super_errors:0|malloc_errors:0|uncorrectable_errors:0|corrected_errors:0|last_physical:0|t_start:1499061605|t_resumed:0|duration:0|canceled:0|finished:1
20d37f57-4cf4-4528-afb8-c62527fb15fe:5|data_extents_scrubbed:51995675|tree_extents_scrubbed:275352|data_bytes_scrubbed:3306572922880|tree_bytes_scrubbed:4511367168|read_errors:0|csum_errors:0|verify_errors:0|no_csum:203072|csum_discards:0|super_errors:0|malloc_errors:0|uncorrectable_errors:0|corrected_errors:0|last_physical:3437686161408|t_start:1499061605|t_resumed:0|duration:19789|canceled:0|finished:1
20d37f57-4cf4-4528-afb8-c62527fb15fe:6|data_extents_scrubbed:51995675|tree_extents_scrubbed:275352|data_bytes_scrubbed:3306572922880|tree_bytes_scrubbed:4511367168|read_errors:0|csum_errors:0|verify_errors:0|no_csum:203072|csum_discards:0|super_errors:0|malloc_errors:0|uncorrectable_errors:0|corrected_errors:0|last_physical:3437686161408|t_start:1499061605|t_resumed:0|duration:19786|canceled:0|finished:1

**Comment by phillxnet** [comment]
@wgriffa757 Thank you for your report. This is a known programmatic issue inadvertently introduced by a necessary API change in our latest release. Please see the following forum thread for context:

https://forum.rockstor.com/t/scheduled-scrub-throws-an-error/3448

From that thread we have the canonical GitHub issue #1759 that your issue is a duplicate of. And the pull request that represents the fix is #1760 which is currently awaiting review and hopefully should be merge soon. Now that I have referenced those GitHub elements this issue is linked to them.

Thanks again for your report and hopefully this should be fixed in our next release. 

Only scheduled scrubs are affected and once the fix is in place your existing scheduled scrubs will have to be recreated. Apologies for any inconvenience.

**Comment by MFlyer** [comment]
duplicate #1762

**Comment by MFlyer** [comment]
Duplicate of #1762

**Comment by phillxnet** [comment]
@MFlyer surely if this issue was a duplicate of itself "Duplicate of #1762" we would be in trouble :). Presumably you mean the #1759 as I indicated previously. Or am I missing something.

**Comment by MFlyer** [comment]
Hi @phillxnet ,
was playing with new Duplicate of tag, but on wrong issue eheh
